src/fal/planner/plan.py

```diff
@@ -28,8 +29,17 @@ def from_execution_plan(
     ) -> FilteredGraph:
         graph = origin_graph.copy_graph()
```

Comment:
Just an idea: Would we want to instead return a filtered view of the original graph? https://networkx.org/documentation/stable/reference/generated/networkx.classes.function.subgraph_view.html?highlight=filter_node

Reply:
Ah, one catch though:

> [subgraph_view](https://networkx.org/documentation/stable/reference/generated/networkx.classes.function.subgraph_view.html?highlight=filter_node#networkx.classes.function.subgraph_view) provides a read-only view of the input graph that excludes nodes and edges based on the outcome of two filter functions filter_node and filter_edge.

We can't use it directly, since it is a read-only view (e.g., in the planner we do in-place mutations).